configs/platforms/windowsfips-2012r2-x64.rb

```diff
@@ -14,7 +14,7 @@
 
   #FIXME we need Fips Compliant Wix, currently not in choco repositories
   #plat.provision_with "C:/ProgramData/chocolatey/bin/choco.exe install -y Wix310 -version 3.10.2 -debug -x86 --no-progress"
-  plat.provision_with "curl -L -o /tmp/wix314-binaries.zip https://artifactory.delivery.puppetlabs.net/artifactory/generic/buildsources/windows/wix3/wix3.14.0.6526-binaries.zip && \"C:/Program Files/7-Zip/7z.exe\" x -y -o\"C:/Program Files (x86)/WiX Toolset v3.14/bin\" C:/cygwin64/tmp/wix314-binaries.zip && rm /tmp/wix314-binaries.zip && SETX WIX \"C:\\Program Files (x86)\\WiX Toolset v3.14\" /M"
+  plat.provision_with "curl -L -o /tmp/wix314-binaries.zip https://artifactory.delivery.puppetlabs.net/artifactory/generic__buildsources/buildsources/wix314-binaries.zip && \"C:/Program Files/7-Zip/7z.exe\" x -y -o\"C:/Program Files (x86)/WiX Toolset v3.14/bin\" C:/cygwin64/tmp/wix314-binaries.zip && rm /tmp/wix314-binaries.zip && SETX WIX \"C:\\Program Files (x86)\\WiX Toolset v3.14\" /M"
```

Comment:
Noticed this after merging, but it's a good idea to pass `--fail` and `--retry`.

The `fail` option is important because if artifactory's response with non-20x, then curl will save the HTML response as the output file!!, e.g. "<html>...<p>Internal Server Error...", return exit code 0 (success). Then we try to unzip the html file, which fails in ways that are hard to troubleshoot.

The `retry` option is good so that we're more resilient to transient network issues.

Reply:
Thanks @joshcooper,
I created https://github.com/puppetlabs/puppet-agent/pull/2558 for this and add the flags to the 7x PR as well: https://github.com/puppetlabs/puppet-agent/pull/2556/files#diff-6009cb070484a9435e8b1bc180e82a88330f040e8f8cd0014510c8e56beb8cdcR17